tests/plugins/repeat_commands.test.ts

```diff
@@ -0,0 +1,504 @@
+  model = new Model();
+  sheetId = model.getters.getActiveSheetId();
+});
+
+describe("Repeat commands basics", () => {
+  test("Repeatable core command list", () => {
+    const repeatableCommands = [
+      "UPDATE_CELL",
+      "CLEAR_CELL",
+      "DELETE_CONTENT",
+      "ADD_MERGE",
+      "REMOVE_MERGE",
+      "SET_FORMATTING",
+      "CLEAR_FORMATTING",
+      "SET_BORDER",
+      "CREATE_FILTER_TABLE",
+      "REMOVE_FILTER_TABLE",
+      "ADD_COLUMNS_ROWS",
+      "REMOVE_COLUMNS_ROWS",
+      "HIDE_COLUMNS_ROWS",
+      "CREATE_SHEET",
+      "DUPLICATE_SHEET",
+      "CREATE_FIGURE",
+      "CREATE_CHART",
+      "CREATE_IMAGE",
+      "HIDE_SHEET",
+      "RESIZE_COLUMNS_ROWS",
+    ].sort();
+    const registryKeys = repeatCommandTransformRegistry.getKeys().sort();
+    expect(repeatableCommands).toEqual(registryKeys);
+  });
+
+  test("Repeatable local command list", () => {
+    const repeatableCommands = [
+      "STOP_EDITION",
+      "PASTE",
+      "INSERT_CELL",
+      "DELETE_CELL",
+      "AUTORESIZE_COLUMNS",
+      "AUTORESIZE_ROWS",
+      "SORT_CELLS",
+      "SUM_SELECTION",
+      "SET_DECIMAL",
+    ].sort();
+    const registryKeys = repeatableLocalCommandsRegistry.getKeys().sort();
+    expect(repeatableCommands).toEqual(registryKeys);
+  });
+
+  test("Can repeat a command", () => {
+    setCellContent(model, "A1", "hello");
+    setSelection(model, ["B2"]);
+    redo(model);
+    expect(getCellContent(model, "B2")).toBe("hello");
+  });
+
+  test("Can undo repeated command", () => {
+    setCellContent(model, "A1", "hello");
+    setSelection(model, ["B2"]);
+    redo(model);
+    expect(getCellContent(model, "B2")).toBe("hello");
+
+    undo(model);
+    expect(getCellContent(model, "B2")).toBe("");
+  });
+
+  test("Can undo => redo => repeat command", () => {
+    setCellContent(model, "A1", "hello");
+    undo(model);
+    expect(getCellContent(model, "A1")).toBe("");
+
+    redo(model);
+    expect(getCellContent(model, "A1")).toBe("hello");
+
+    setSelection(model, ["B2"]);
+    redo(model);
+    expect(getCellContent(model, "B2")).toBe("hello");
+  });
+});
+
+describe("Repeat command transform generics", () => {
+  test.each([
+    TEST_COMMANDS.UPDATE_CELL,
+    TEST_COMMANDS.CLEAR_CELL,
+    TEST_COMMANDS.DELETE_CONTENT,
+    TEST_COMMANDS.ADD_MERGE,
+    TEST_COMMANDS.REMOVE_MERGE,
+    TEST_COMMANDS.SET_FORMATTING,
+    TEST_COMMANDS.CLEAR_FORMATTING,
+    TEST_COMMANDS.SET_BORDER,
+    TEST_COMMANDS.CREATE_FILTER_TABLE,
+    TEST_COMMANDS.REMOVE_FILTER_TABLE,
+    TEST_COMMANDS.HIDE_SHEET,
+  ])("Sheet dependant command are adapted to current sheet %s", (command: CoreCommand) => {
+    createSheet(model, { sheetId: "42" });
+    activateSheet(model, "42");
+    const transformed = getRepeatedCoreCommands(model.getters, [command])[0];
+    expect(transformed?.sheetId).toEqual("42");
+  });
+
+  test.each([TEST_COMMANDS.UPDATE_CELL!, TEST_COMMANDS.CLEAR_CELL!, TEST_COMMANDS.SET_BORDER])(
```

Comment:
Why have you kept these "!" for UPDATE_CELL and CLEAR_CELL ? Is there something special with these commands ?